### ISSUE TITLE
datatables view search behaviour 

### ISSUE THREAD

**Issue by jean** [plomino/Plomino#596]
I'm observing some behaviour that will probably be surprising and undesirable for most users.
When searching a datatable, the search is sent back to `PlominoView.tojson`, which calls `getAllDocuments` with a `fulltext_query` parameter.

This means that the search doesn't correspond to the visible information in the view table. E.g. if there is a status column showing "Active/Inactive", then a search on *Inactive* will come up blank unless this status information is cataloged as part of the view documents' `SearchableText`. 

I propose searching clientside (like datagrid fields) first, and then adding results from a serverside search. Would this work?

**Comment by ebrehault** [comment]
I agree the behaviour is confusing, but returning both clientside results plus serverside results would even more confusing.
Originally, we were just using clientside results, but it was not accurate when the view is paginated.
That's why we swicthed to serverside, but only using `SearchableText`.
The ideal solution would be to search serverside but using the view's columns indexes.

**Comment by jean** [comment]
> The ideal solution would be to search serverside but using the view's
columns indexes.

That sounds doable .. I'll have a look.

-- 
J

**Comment by AshRaghav** [comment]
Thanks for bringing this up as we face a similar issue. The datatable search box is searching inside all fields inside a document and that returns inaccurate results because the search box is meant to show results that match the data from visible columns in a table. 

Is there an update to this implementation? Can we restrict the datatable search to specific columns in a view?

Thanks again!

**Comment by jean** [comment]
Maybe you can look at datatables for an answer, e.g. https://datatables.net/reference/option/searchCols
You could configure the view as the data source for a datagrid field.
Then on the Plomino side, that config goes here: http://plomino.readthedocs.io/en/latest/fields/#datagrid-field

This probably depends on the size of the dataset.

**Comment by AshRaghav** [comment]
Thanks Jean. 

I did have a look at searchCols before but that will work only during table initialisation. Maybe I haven't explained my issue correctly.

What I am looking for is a View (with dynamic table) and the search option (available by default) to search within specific visible columns on the table rather than searching through all fields inside the document. 

Please correct me if I misunderstood you previously. I'll go back and investigate further.
